naga/src/front/wgsl/parse/lexer.rs

```diff
@@ -81,28 +83,45 @@ fn consume_token(input: &str, generic: bool) -> (Token<'_>, &str) {
             let og_chars = chars.as_str();
             match chars.next() {
                 Some('/') => {
-                    let _ = chars.position(is_comment_end);
-                    (Token::Trivia, chars.as_str())
+                    if let Some(end_position) = input
+                        .char_indices()
+                        .find(|char_indices| is_comment_end(char_indices.1))
+                    {
+                        let end_position = end_position.0 + 1;
+                        return (
+                            if chars.next() == Some('!') {
+                                Token::CommentModule(&input[..end_position])
+                            } else {
+                                Token::Comment(&input[..end_position])
+                            },
+                            &input[end_position..],
+                        );
+                    }
+                    (Token::Comment(input), &input[input.len() - 1..])
```

Comment:
This doesn't look right. Doesn't it just return the last byte of `input` as the following text?

This code is only exercised by `front::wgsl::parse::lexer::test_comments`, which only uses a `//` at the end of the input, so it wouldn't notice if the rest of the text were truncated.